addons/udes_warehouse_classification/tests/test_product_template.py

```diff
@@ -0,0 +1,59 @@
+from .common import Base
+
+
+class TestWarehouseClassificationPicking(Base):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.test_messages_01 = ["Inbound, Report a, Report"]
+        cls.test_messages_02 = ["Report a Report", "Report b Report"]
+        cls.test_messages_03 = ["Report a Report", "Inbound, Report a, Report"]
+
+    # functions to test:  and _get_classification_messages_for_product_picking
+
+    # For a given product find the classification messages for specified report,
+    # 1. report a on it's own - banana
+    # 2. not matching report type - banana
+    # 3. report a and b - honeydew
+    # 4. report a and report a on inbound (i.e. 2 messages for a report) - cherry
+    # 5. no classifications - apple
+
+    def test01_basic_report(self):
+        messages = self.banana.product_tmpl_id.get_classification_messages_for_report(
```

Comment:
why `banana.product_tmpl_id.get_classification_messages_for_report` and not just `banana.get_classification_messages_for_report`?

Reply:
made a proxy method from product.product